layouts/partials/clickable.html

```diff
@@ -0,0 +1,29 @@
+{{- $open_tag := "<button" -}}
+{{- $close_tag := "</button>" -}}
+{{- if .link -}}
```

Comment:
I would use the [cond](https://gohugo.io/functions/cond/) function here. Maybe something like this.

```
{{ $tag := (cond .link "a" "button") }}
<{{ $tag }} ... >
</{{ $tag }}>
```

You might need to do `(isset . "link")` instead of just `.link` I can't remember off the top of my head.

Reply:
This syntax doesn't work:

![image](https://user-images.githubusercontent.com/8326331/119431147-0e16bf00-bce0-11eb-9fa4-b6d2b8a24d5b.png)

See this link: https://discourse.gohugo.io/t/using-custom-variables-for-html-tags/15034/7 (Side note, I really despise the moderator saying "Would you rather “readable” or “working” :slight_smile: ?")

And also this was one of my bullet points in my note above.

I did change the code to make use of `cond` though.